src/clients/AcrossConfigStoreClient.ts

```diff
@@ -0,0 +1,436 @@
+      }
+    }
+
+    // Save new Global config updates.
+    for (let i = 0; i < updatedGlobalConfigEvents.length; i++) {
+      const event = updatedGlobalConfigEvents[i];
+      const args = {
+        blockNumber: event.blockNumber,
+        transactionIndex: event.transactionIndex,
+        logIndex: event.logIndex,
+        ...spreadEvent(event.args),
+      };
+
+      if (args.key === utf8ToHex(GLOBAL_CONFIG_STORE_KEYS.MAX_RELAYER_REPAYMENT_LEAF_SIZE)) {
+        if (!isNaN(args.value)) {
+          this.cumulativeMaxRefundCountUpdates.push(args);
+        }
+      } else if (args.key === utf8ToHex(GLOBAL_CONFIG_STORE_KEYS.MAX_POOL_REBALANCE_LEAF_SIZE)) {
+        if (!isNaN(args.value)) {
+          this.cumulativeMaxL1TokenCountUpdates.push(args);
+        }
+      } else if (args.key === utf8ToHex(GLOBAL_CONFIG_STORE_KEYS.VERSION)) {
+        // If not a number, skip.
+        if (isNaN(args.value)) {
+          continue;
+        }
+        const value = Number(args.value);
+
+        // If not an integer, skip.
+        if (!Number.isInteger(value)) {
+          continue;
+        }
+
+        // Extract last version
+        const lastValue =
+          this.cumulativeConfigStoreVersionUpdates.length === 0
+            ? DEFAULT_CONFIG_STORE_VERSION
+            : Number(
+                this.cumulativeConfigStoreVersionUpdates[this.cumulativeConfigStoreVersionUpdates.length - 1].value
+              );
+
+        // If version is not > last version, skip.
+        if (value <= lastValue) {
+          continue;
+        }
+
+        this.cumulativeConfigStoreVersionUpdates.push({
+          ...args,
+          timestamp: globalConfigUpdateTimes[i],
+        });
+      } else if (args.key === utf8ToHex(GLOBAL_CONFIG_STORE_KEYS.DISABLED_CHAINS)) {
+        try {
+          const chainIds = this.filterDisabledChains(JSON.parse(args.value) as number[]);
+          this.cumulativeDisabledChainUpdates.push({ ...args, chainIds });
+        } catch (err) {
+          // Can't parse list, skip.
+        }
+      } else {
+        continue;
+      }
+    }
+
+    this.rateModelDictionary.updateWithEvents(this.cumulativeRateModelUpdates);
+
+    this.hasLatestConfigStoreVersion = this.hasValidConfigStoreVersionForTimestamp();
+    this.isUpdated = true;
+    this.firstBlockToSearch = searchConfig.toBlock + 1; // Next iteration should start off from where this one ended.
+
+    this.logger.debug({ at: "ConfigStore", message: "ConfigStore client updated!" });
+  }
+
+  filterDisabledChains(disabledChains: number[]): number[] {
+    // If any chain ID's are not integers then ignore. UMIP-157 requires that this key cannot include
+    // the chain ID 1.
+    return disabledChains.filter((chainId: number) => !isNaN(chainId) && Number.isInteger(chainId) && chainId !== 1);
+  }
+
+  private async getBlockNumber(timestamp: number): Promise<number | undefined> {
+    return (await this.blockFinder.getBlockForTimestamp(timestamp)).number;
+  }
```

Comment:
Replaced the code with the below. In the `relayer-v2` repository, I plan to overload this function in the extended child class.

```
  private async getBlockNumber(timestamp: number): Promise<number | undefined> {
    return await getBlockForTimestamp(
      this.hubPoolClient.chainId,
      this.hubPoolClient.chainId,
      timestamp,
      getCurrentTime(),
      this.blockFinder
    );
  }
```